python-sdk/tests/test_example_dags.py

```diff
@@ -51,44 +51,44 @@ def session():
     return get_session()
 
 
-DAG_BAG = DagBag(Path(__file__).parent.parent / "example_dags", include_examples=False)
-AIRFLOW_VERSION_INDICATOR = "airflow_version:"
-MINIMUM_AIRFLOW_VERSION = "2.2.5"
+MIN_VER_DAG_FILE: dict[str, list[str]] = {
+    "2.3": ["example_dynamic_task_template.py", "example_bigquery_dynamic_map_task.py"],
+    "2.4": ["example_datasets.py"],
+}
 
+# Sort descending based on Versions and convert string to an actual version
+MIN_VER_DAG_FILE_VER: dict[Version, list[str]] = {
+    Version(version): MIN_VER_DAG_FILE[version]
+    for version in sorted(MIN_VER_DAG_FILE, key=Version, reverse=True)
+}
 
-def get_airflow_version(dag: DAG) -> str:
-    for tag in dag.tags:
-        if tag.startswith(AIRFLOW_VERSION_INDICATOR):
-            return tag[len(AIRFLOW_VERSION_INDICATOR) :]
-    return MINIMUM_AIRFLOW_VERSION
 
+def get_dag_bag() -> DagBag:
+    """Create a DagBag by adding the files that are not supported to .airflowignore"""
+    example_dags_dir = Path(__file__).parent.parent / "example_dags"
+    airflow_ignore_file = example_dags_dir / ".airflowignore"
 
-def get_airflow_dags() -> Iterator[tuple[str, str]]:
-    for dag_id, dag in DAG_BAG.dags.items():
-        yield dag_id, dag, get_airflow_version(dag)
+    with open(airflow_ignore_file, "w+") as file:
+        for min_version, files in MIN_VER_DAG_FILE_VER.items():
+            if Version(airflow.__version__) < min_version:
+                print(f"Adding {files} to .airflowignore")
+                file.writelines([f"{file}\n" for file in files])
+
+    print(".airflowignore contents: ")
+    print(airflow_ignore_file.read_text())
+    dag_bag = DagBag(example_dags_dir, include_examples=False)
+    return dag_bag
 
 
 @pytest.mark.parametrize(
     "dag",
-    [
-        pytest.param(
-            dag,
-            id=dag_id,
-            marks=pytest.mark.skipif(
-                airflow.__version__ < dag_airflow_version,
-                reason=f"Require Airflow version >= {dag_airflow_version}",
-            ),
-        )
-        for dag_id, dag, dag_airflow_version in get_airflow_dags()
-    ],
+    [pytest.param(dag, id=dag_id) for dag_id, dag in get_dag_bag().dags.items()],
```

Comment:
It is not necessary that you pull the dags. You can use `@pytest.mark.parametrize("dag_id", dag_bag.dag_ids)` and then pull the dag inside the function. This might be a bit faster. WDYT?